streams/src/test/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueLoggedStoreTest.java

```diff
@@ -57,4 +68,147 @@ public void shouldPutAll() {
+    @Test
+    public void shouldReturnAll() {
+        final InMemoryKeyValueLoggedStore<Integer, String> inMemoryKeyValueLoggedStore = getInMemoryLoggedStore(3, integerSerde, stringSerde);
+        inMemoryKeyValueLoggedStore.put(1, "one");
+        inMemoryKeyValueLoggedStore.put(2, "two");
+        inMemoryKeyValueLoggedStore.put(3, "three");
+
+        final List<KeyValue<Integer, String>> allReturned = new ArrayList<>();
+        final List<KeyValue<Integer, String>> expectedReturned = Arrays.asList(KeyValue.pair(1, "one"), KeyValue.pair(2, "two"), KeyValue.pair(3, "three"));
+        final Iterator<KeyValue<Integer, String>> iterator = inMemoryKeyValueLoggedStore.all();
+
+        while (iterator.hasNext()) {
+            allReturned.add(iterator.next());
+        }
+
+        assertThat(allReturned, equalTo(expectedReturned));
+    }
+
+    @Test
+    public void shouldPutAllInLoggedStore() {
+        final InMemoryKeyValueLoggedStore<Integer, String> inMemoryKeyValueLoggedStore = getInMemoryLoggedStore(3, integerSerde, stringSerde);
+        final List<KeyValue<Integer, String>> allReturned = new ArrayList<>();
+        final List<KeyValue<Integer, String>> initialPutAll = Arrays.asList(KeyValue.pair(1, "one"), KeyValue.pair(2, "two"), KeyValue.pair(3, "three"));
+
+        inMemoryKeyValueLoggedStore.putAll(initialPutAll);
+
+        final Iterator<KeyValue<Integer, String>> iterator = inMemoryKeyValueLoggedStore.all();
+
+        while (iterator.hasNext()) {
+            allReturned.add(iterator.next());
+        }
+        assertThat(allReturned, equalTo(initialPutAll));
+    }
+
+    @Test
+    public void shouldPutIfAbsentValidUpdate() {
+        final InMemoryKeyValueLoggedStore<Integer, String> inMemoryKeyValueLoggedStore = getInMemoryLoggedStore(3, integerSerde, stringSerde);
+        String returnedValue = inMemoryKeyValueLoggedStore.putIfAbsent(1, "one");
+        // null not previously in store
+        assertNull(returnedValue);
+
+        String attemptedUpdate = inMemoryKeyValueLoggedStore.putIfAbsent(1, "one-more");
+
+        // already in store so sticking with original value, no update
+        assertThat(attemptedUpdate, equalTo("one"));
+    }
+
+
+    @Test
+    public void shouldReturnRange() {
+        final InMemoryKeyValueLoggedStore<Integer, String> inMemoryKeyValueLoggedStore = getInMemoryLoggedStore(3, integerSerde, stringSerde);
+        final List<KeyValue<Integer, String>> allReturnedRange = new ArrayList<>();
+        final List<KeyValue<Integer, String>>
+            initialPutAll =
+            Arrays.asList(KeyValue.pair(1, "one"), KeyValue.pair(2, "two"), KeyValue.pair(3, "three"), KeyValue.pair(4, "four"));
+
+        final List<KeyValue<Integer, String>> expectedRange = Arrays.asList(KeyValue.pair(3, "three"), KeyValue.pair(4, "four"));
+        inMemoryKeyValueLoggedStore.putAll(initialPutAll);
+
+        final Iterator<KeyValue<Integer, String>> iterator = inMemoryKeyValueLoggedStore.range(3, 4);
```

Comment:
Should we check different ranges? For example, range that is completely outside (before and after) available range? Range that is overlapping (2x). Range that is completely contained? Range that return everything?

Reply:
I'm going to update test with a mock.